doc/build-helpers/trivial-build-helpers.chapter.md

```diff
@@ -92,6 +92,155 @@ writeShellScript "evaluate-my-file.sh" ''
 ```
 ::::
 
+### `makeDesktopItem` {#trivial-builder-makeDesktopItem}
+
+Write an [XDG desktop file](https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html) to the Nix store.
+
+This function is most often used with `mkDerivation.desktopItems` to add desktop items to a package.
+
+`makeDesktopItem` takes an attribute set according to the [XDG specification](https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#recognized-keys), with the following possible values:
+
+Note that the 'Version' and 'Hidden' fields are not supported.
+
+`name` (String)
+
+: The name of the desktop file in the Nix store.
+
+`type` (String, _optional_)
+
+: According to the specification.
+
+  Default: `"Application"`
+
+`desktopName` (String)
+
+: According to the specification.
+
+`genericName` (String, _optional_)
+
+: According to the specification.
+
+`noDisplay` (Bool, _optional_)
+
+: According to the specification.
+
+`comment` (String, _optional_)
+
+: According to the specification.
+
+`icon` (String, _optional_)
+
+: According to the specification.
+
+`onlyShowIn` (List of String, _optional_)
+
+: According to the specification.
+
+`notShowIn` (List of String, _optional_)
+
+: According to the specification.
+
+`dbusActivatable` (Bool, _optional_)
+
+: According to the specification.
+
+`tryExec` (String, _optional_)
+
+: According to the specification.
+
+`exec` (String, _optional_)
+
+: According to the specification.
+
+`path` (String, _optional_)
+
+: According to the specification.
+
+`terminal` (Bool, _optional_)
+
+: According to the specification.
+
+`actions` (List of Attribute set, _optional_)
+
+: According to the specification.
+
+  A list of attribute sets {name, exec?, icon?}
+
+`mimeTypes` (List of String, _optional_)
+
+: According to the specification.
+
+`categories` (List of String, _optional_)
+
+: According to the specification.
+
+`implements` (List of String, _optional_)
+
+: According to the specification.
+
+`keywords` (List of String, _optional_)
+
+: According to the specification.
+
+`startupNotify` (Bool, _optional_)
+
+: According to the specification.
+
+`startupWMClass` (String, _optional_)
+
+: According to the specification.
+
+`url` (String, _optional_)
+
+: According to the specification.
+
+`prefersNonDefaultGPU` (Bool, _optional_)
+
+: According to the specification.
+
+  Note that this is not supported until version 1.5, which the `desktop-file-utils` in Nixpkgs does not support.
+
+`extraConfig` (Attribute set, _optional_)
+
+: Additional key/value pairs to be added verbatim to the desktop file. Attributes need to be prefixed with 'X-'.
+
+::: {.example #ex-makeDesktopItem}
+# Usage of `makeDesktopItem`
+
+Write a desktop file `/nix/store/<store path>/my-program.desktop` to the Nix store.
```

Comment:
Ideally it would be nicer to have an example that would work if someone decided to run this and use it - would it be possible to craft an example for the `hello` package, for example? The example doesn't need to use all the properties, but if someone decides to use the output to actually run the `hello` package, things should work.

If not, we can leave this generic example.